637106: Outreach for article clean-up: chgranad (2)

vatmicro opened the pull request:
This PR is created for the following [user story](https://dev.azure.com/ceapex/Contributor%20Success/_workitems/edit/637106)

PRMerger5 commented:
@vatmicro : Thanks for your contribution! The author(s) have been notified to review your proposed change.

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -2,59 +2,58 @@
+We recommend:
 
-- Never disregard a convention unless you’re doing so intentionally in order to provide more readable and understandable code for your users.
+- Stick to convention unless you intentionally want to provide code that is more readable and understandable for users.
```
I'm not sure I entirely agree with this edit; makes it sound like breaking convention is _in general_ a way to provide more readable and understandable code, but that's the exception rather than the norm.

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -2,59 +2,58 @@
-To communicate the distinction between functions and operations to users, we rely on that Q# models quantum operations through the use of side effects.
+Our name choices should first establish whether a symbol represents a function or an operation.
+The difference between functions and operations indicates how a block of code behaves.
+We rely on Q# models quantum operations to communicate the difference between functions and operations to users, through the use of side effects.
```
```suggestion
We rely on how Q# models quantum operations to communicate the difference between functions and operations to users, through the use of side effects.
```

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -2,59 +2,58 @@
 
-One case that deserves special mention is when an operation takes another operation as input and calls it.
-In such cases, the action taken by the input operation is not clear when the outer operation is defined, such that the right verb is not immediately clear.
+We should pay special attention when an operation takes another operation as input. Also, when the operation calls it.
```
I think this edit changes the meaning of the original text, making it sound like "takes another operation" and "calls it" are two different cases rather than two steps in a single case.

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -108,16 +107,16 @@ We suggest:
-When defining an entry point into a Q# program, the Q# compiler recognizes the [`@EntryPoint()` attribute](xref:Microsoft.Quantum.Core.EntryPoint) rather requiring that entry points have a particular name (for example: `main`, `Main`, or `__main__`).
-That is, from the perspective of a Q# developer, entry points are ordinary operations annotated with `@EntryPoint()`.
-Moreover, Q# entry points may be entry points for an entire application (for example, in Q# standalone executable programs), or may be an interface between a Q# program and the host program for an application (for example, when using Q# with Python or .NET), such that the name "main" could be misleading when applied to a Q# entry point.
+When you define entry points for Q# programs, the Q# compiler recognizes the [`@EntryPoint()` attribute](xref:Microsoft.Quantum.Core.EntryPoint). Entry points should have a specifc name (for example: `main`, `Main`, or `__main__`).
```
This edit inverts the meaning of the original text; Q# explicitly does not require naming entry points as `main`, `Main`, or `__main__`.

cgranade replied:
In languages like Python, C, C++, Java, and C#, it is typical that the entry point for an application is identified by a special name like `main`, `Main`, or `__main__`. That in turn requires that users name their entry points in a special way.

Q#, by contrast, does not impose such a requirement. Entry points are denoted by `@EntryPoint`, but users are free to name those entry points whatever they want. The style guide recommends that users do so in a way that is consistent with more general guidelines for naming Q# operations (as verbs, avoiding esoteric names, and so forth).

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -108,16 +107,16 @@ We suggest:
+Moreover, Q# entry points may be entry points for an entire application (for example, in Q# standalone executable programs), or may be an interface between a Q# program and the host program for an application (for example, when you use Q# with Python or .NET), such that the name "main" could be misleading when applied to a Q# entry point.
 
-We suggest using naming entry points to reflect the use of the `@EntryPoint()` attribute by using the general advice for naming operations listed above.
+We recommend that you use naming entry points that reflect the use of the `@EntryPoint()` attribute by using the general advice for naming operations listed above.
```
Is this gramatically correct?

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -181,25 +180,24 @@ We suggest:
-In both cases, the API documentation comments provide synonymous names based on proper nouns, along with all appropriate citations.
-
-This preference is especially important given that some usage of proper nouns will always be necessary — Q# follows the tradition set by many classical languages, for instance, and refers to `Bool` types in reference to Boolean logic, which is in turn named in honor of George Boole.
+In physics it is common to name things after the first person that published information about it. HOever, most non-physicists aren’t familiar with the important names and history of physics.
```
This looks to be a spelling error? The edit here also changes the meaning of the original text by adding "important;" the point of this paragraph is that memorizing physics history should _not_ be important to the act of writing a quantum program.

vatmicro replied:
Pls review the re-write when you are notified by GitHub of the next build.

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -181,25 +180,24 @@ We suggest:
+As an example, the singly controlled SWAP and doubly controlled NOT operations are often called the "Fredkin" and "Toffoli" operations in academic literature, but are identified in Q# primarily as `CSWAP` and `CCNOT`.
+In both cases, API documentation comments provide synonymous names based on proper nouns, along with all appropriate citations.
+
+This option is especially important since some use of proper nouns will always be necessary — Q# follows the tradition set by many classical languages, for instance, and refers to `Bool` types in reference to Boolean logic, which is in turn named in honor of George Boole.
```
"Option" is weaker than "preference," making it sound like this paragraph can be safely ignored.

vatmicro replied:
@cgranade - Both of them imply a choice.  Is this a preference or a definite instruction?

cgranade replied:
Agreed, both represent choices, but "option" makes it sound like a completely neutral choice, undermining the above guidance. It is a preference that authors can override based on their best judgment, but that they should follow by default.

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -181,25 +180,24 @@ We suggest:
+We recommend:
 
-- Avoid the use of proper nouns in names.
+- Avoid use of proper nouns in names.
```
Using?

vatmicro replied:
When we make this sentence active, it becomes, "Don't use proper nouns when you choose names".

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -234,15 +232,15 @@ We suggest:
 ### Private or internal names ###
 
-In many cases, a name is intended strictly for use internal to a library or project, and is not a guaranteed part of the API offered by a library.
+In many cases, a name is intended strictly for internal use in a library or project. It is not a guaranteed part of the API that a library offers.
```
This edit seems to change the meaning as well?

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -234,15 +232,15 @@ We suggest:
+In many cases, a name is intended strictly for internal use in a library or project. It is not a guaranteed part of the API that a library offers.
 It is helpful to clearly indicate that this is the case when naming functions and operations so that accidental dependencies on internal-only code are made obvious.
-If an operation or function is not intended for direct use, but rather should be used by a matching callable which acts by partial application, consider using a name starting with the `internal` keyword for the callable that is partially applied.
+There are operations and functions that aren't intended for direct use. They can be used by a matching callable which acts by partial application. Then consider using a name that starts with the `internal` keyword for the callable that is partially applied.
```
The existence of such functions and operations is already considered at the start of the paragraph; this sentence is intended to provide guidance for specific examples of such functions and operations.

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -234,15 +232,15 @@ We suggest:
+We recommend:
 
-- When a function, operation, or user-defined type is not a part of the public API for a Q# library or program, ensure that it is marked as internal by placing the `internal` keyword before the `function`, `operation`, or `newtype` declaration.
+- When functions, operations, or user-defined types are not part of the public API for a Q# library or program, mark it as internal by placing the `internal` keyword before the `function`, `operation`, or `newtype` declaration.
```
Subject–verb agreement? Should "mark it" be "mark them"?

vatmicro replied:
Correct, I have made the change.

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -328,16 +326,16 @@ We suggest:
-Thus, partially applying the first arguments should result in a callable that is useful in its own right whenever that is reasonable.
-Following this principle, consider using the following order of arguments:
+We recommend an input order convention. Create this by thinking of partial application as a generalization of currying 𝑓(𝑥, 𝑦) ≡ 𝑓(𝑥)(𝑦).
+Partially apply the first arguments which results in a callable that is useful in its own right whenever that is reasonable.
```
This changes the meaning to instruct readers to partially apply functions and operations rather than to consider the experience their users will have when they partially apply.

vatmicro replied:
I have reverted this statement.

cgranade commented on articles/contributing-style-guide.md:
```diff
@@ -360,16 +358,16 @@ operation ApplyPhaseEstimationIteration(
+Some functions and operations mimic the behavior of the built-in functors `Adjoint` and `Controlled`.
 For instance, `ControlledOnInt<'T>` has type `(Int, ('T => Unit is Adj + Ctl)) => ((Qubit[], 'T) => Unit is Adj + Ctl)`, such that `ControlledOnInt<Qubit[]>(5, _)` acts like the `Controlled` functor, but on the condition that the control register represents the state $\ket{5} = \ket{101}$.
-Thus, a developer expects that the inputs to `ControlledOnInt` place the callable being transformed last, and that the resulting operation takes as its input `(Qubit[], 'T)` --- the same order as followed by the output of the `Controlled` functor.
+Developers expect that input to `ControlledOnInt` places the callable that is being transformed, last in the queue. They also assume that the resulting operation accepts `(Qubit[], 'T)` as its input--- the same order as followed by the output of the `Controlled` functor.
```
Input should still be pluralized here.

vatmicro replied:
I have pluralized inputs again

cgranade commented on articles/user-guide/libraries/chemistry/schema/spec_v_0_1.md:
```diff
@@ -1,29 +1,30 @@
 Any sidebar with the headings "NOTE," "INFORMATION," or "WARNING" is informative.
 
-## Introduction ##
+## What are Broombridge Documents? ##
```
This document is a formal specification, such that the headers are intended to be as previously written.

vatmicro replied:
Ill revert this heading

cgranade commented on articles/user-guide/libraries/standard/math.md:
```diff
@@ -1,20 +1,21 @@
+These functions primarily work with Q# built-in data types `Int`, `Double`, and `Range`.
 
-For example, the <xref:Microsoft.Quantum.Random.DrawCategorical> has signature `(Double[] => Int)`. It takes a categorical distribution specified by a list of probablities, such as an array of doubles as input, and returns a randomly-selected index into the array as an `Int`.
+As an example, the <xref:Microsoft.Quantum.Random.DrawCategorical> has the signature `(Double[] => Int)`. It takes a categorical distribution specified by a list of probablities. An array of doubles can be entered as an input, and then return a randomly-selected index into the array as an `Int`.
```
This feels really awkward now?

vatmicro replied:
Hi Cassandra, I understand it may feel a little awkward. 

The sentence was taken from passive voice to active voice. I recommend that we keep the change as the style guide advises that we use active voice wherever possible.

cgranade reviewed (commented):
Thanks for your edits! I've highlighted some places where the changes in this PR may change the original and intended meaning of the documentation pages in question; please let me know if there's anything I can do to clarify or expand on my comments. Thank you!

vatmicro commented:
@cgranade , thank you for the feedback. I will address the above and resubmit.

Court72 commented:
Hi @vatmicro, @cgranade,

These commits were made in the *public* repo. Should they be made in the *private* repo instead? 

In the private repo, PRs benefit from staging and validation.

Thanks!
#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

vatmicro commented:
> Hi @vatmicro, @cgranade,
> 
> These commits were made in the _public_ repo. Should they be made in the _private_ repo instead?
> 
> In the private repo, PRs benefit from staging and validation.
> 
> Thanks! #label:"aq-pr-triaged" @MicrosoftDocs/public-repo-pr-review-team

Yes, I believe so.

vatmicro commented:
HI @cgranade , I have addressed the issues you pointed out above in various ways, however I need more information from you to resolve a few of the issues. Could you please supply the additional info that I need? Thank you.

PRMerger12 commented:
@vatmicro : Thanks for your contribution! The author(s) have been notified to review your proposed change.

vatmicro commented:
Hi @cgranade , I have addressed all of your question with this build latest build, aside from those that I have asked for extra clarification on.  
Please also note that our goal is to resolve all issues and receive your approval on the PR by end of day tomorrow June 29, since the project is coming to an end.  
Thank you.
